### DISCUSSION TITLE
Can Typescript Playwright tests be used for Artillery perf tests?

### DISCUSSION THREAD

**Discussion by Icaroto** [artilleryio/artillery#1882] [Artillery]
Hi,

Really sorry if the question is not relevant or already answered (I couldn't find any), I'm looking for a performance tool and came across Artillery  today which seems fantastic, specially with the Playwright integration! 

Considering our  Playwright tests are all in Typescript, is it still possible to use Artillery to run them? If so, is there any example to follow?

Thanks!

**Comment by hassy** [comment]
Hi @Icaroto 👋 Not at the moment, though we do want to add TypeScript support in future. Right now only plain JS is supported.

**Comment by TenSt** [comment]
Hi @hassy 👋 can you share any plans or roadmap of when you plan to add TypeScript support? 🙏

**Comment by sanisc** [comment]
is the mentioned in the docs anywhere? if so it should be.

**Reply by Tommacfie** [reply]
100%, spent a lot of time trying to integrate Artillery into my current TS Playwright setup, which I had to give up on once I read this thread and realised. Would be nice to know since TS is becoming much more widely adopted and I would just assume that the support would be in

**Reply by hassy** [reply]
Thanks folks! Just added a note on current limitations in Playwright support in the docs - https://www.artillery.io/docs/reference/engines/playwright#current-limitations 

We do intend on adding support for TS. (It's not a Playwright-specific feature/limitation. Artillery itself only supports CommonJS modules. Once we've added support for TS, it will just work with Playwright.)

**Reply by thomasdavis** [reply]
You could just convert your modules to es before running a test

**Comment by bernardobridge** [comment]
Hey folks 👋

Just an update here. The new Artillery release ([v2.0.6](https://github.com/artilleryio/artillery/releases/tag/artillery-2.0.6)) contains improvements to the experimental Typescript support (available since v2.0.4). You should now be able to use expect from @playwright/test within your test scripts.

Some documentation:
- https://www.artillery.io/docs/reference/test-script#preventing-bundling-of-typescript-packages
- https://www.artillery.io/docs/reference/engines/playwright#known-limitations
